lib/rubygems-requirements-system/platform/msys2.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2025  Ruby-GNOME Project Team
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+require_relative "base"
+
+module RubyGemsRequirementsSystem
+  module Platform
+    class MSYS2 < Base
+      Platform.register(self)
+
+      class << self
+        def current_platform?
+          return false if Object.const_defined?(:RubyInstaller)
+          return false if package_prefix.nil?
+          not pacman_path.nil?
+        end
+
+        def pacman_path
+          finder = ExecutableFinder.new("pacman")
+          case RUBY_PLATFORM
+          when "i386-mingw32"
+            Dir.glob("c:/msys{64,32,*}/usr/bin").each do |bin|
+              finder.append_path(bin)
+            end
+          when "x64-mingw32", "x64-mingw-ucrt", "aarch64-mingw-ucrt"
+            Dir.glob("c:/msys{64,*}/usr/bin").each do |bin|
+              finder.append_path(bin)
+            end
+          end
+          finder.find
+        end
+
+        def package_prefix
+          case RUBY_PLATFORM
+          when "i386-mingw32"
+            "mingw-w64-i686-"
+          when "x64-mingw32"
+            "mingw-w64-x86_64-"
+          when "x64-mingw-ucrt"
+            "mingw-w64-ucrt-x86_64-"
+          when "aarch64-mingw-ucrt"
+            "mingw-w64-clang-aarch64-"
+          else
+            nil
+          end
+        end
```

Comment:
Let's assume that we are in MSYS2 shell:

```suggestion
          ENV["MINGW_PACKAGE_PREFIX"]
```

Note that `MINGW_PACKAGE_PREFIX` doesn't include the last `-` like `mingw-w64-clang-x86_64`.